docs/02-app/01-building-your-application/09-authentication/index.mdx

```diff
@@ -351,7 +351,7 @@ import type { NextRequest } from 'next/server'
 export function middleware(request: NextRequest) {
   const currentUser = request.cookies.get('currentUser')?.value
 
-  if (currentUser) {
+  if (currentUser && !request.nextUrl.pathname.startsWith('/dashboard')) {
     return NextResponse.redirect(new URL('/dashboard', request.url))
   }
   return NextResponse.redirect(new URL('/login', request.url))
```

Comment:
Do we also need the same guard for `/login` here?

Reply:
Good point! Option 1 is feasible. If we added /login to the matcher to exclude it, it would never run middleware for that route and the redirect would have no effect if the user is logged in and on the login route.